src/lib/eina/eina_thread_win32.c

```diff
@@ -0,0 +1,188 @@
+}
+
+void _eina_thread_set_priority(Eina_Thread_Priority prio, Eina_Thread *t)
+{
+
+   int nPriority;
+
+   switch (prio)
+   {
+   case EINA_THREAD_URGENT:
+      nPriority = THREAD_PRIORITY_HIGHEST;
+      break;
+   case EINA_THREAD_NORMAL:
+      nPriority = THREAD_PRIORITY_NORMAL;
+      break;
+   case EINA_THREAD_BACKGROUND:
+      nPriority = THREAD_PRIORITY_BELOW_NORMAL;
+      break;
+   case EINA_THREAD_IDLE:
+      nPriority = THREAD_PRIORITY_IDLE;
+      break;
+   }
+
+   SetThreadPriority((HANDLE)*t, nPriority);
+}
+
+inline Eina_Bool
+_eina_thread_create(Eina_Thread *t, int affinity, void *(*func)(void *data), void *data)
+{
+   Eina_Bool ret;
+   Eina_win32_thread_attr thread_attr;
+
+   SECURITY_ATTRIBUTES sec_attributes;
+
+   sec_attributes.nLength = sizeof(SECURITY_ATTRIBUTES);
+   sec_attributes.lpSecurityDescriptor = NULL;
+   sec_attributes.bInheritHandle = EINA_TRUE;
+
+   thread_attr.lpThreadAttributes = &sec_attributes;
+
+   thread_attr.dwStackSize = 4000;
+   thread_attr.dwCreationFlags = 0;
+
+   LPDWORD threadID;
+
+   Eina_win32_thread_func thread_func;
+   Eina_Thread_Call *c = (Eina_Thread_Call *)(data);
+
+   thread_func.func = func;
+   thread_func.data = data;
+
+   *t = (HANDLE)CreateThread(thread_attr.lpThreadAttributes, thread_attr.dwStackSize, &_eina_thread_func, &thread_func, thread_attr.dwCreationFlags, threadID);
+
+   _eina_thread_set_priority(c->prio, t);
+
+   ret = (*t != NULL) ? EINA_TRUE : EINA_FALSE;
+
+   if (affinity >= 0 && ret)
+   {
+      SetThreadAffinityMask(*t, (DWORD_PTR *)&affinity);
+   }
+
+   return ret;
+}
+
+inline Eina_Bool
+_eina_thread_equal(Eina_Thread t1, Eina_Thread t2)
+{
+   DWORD t1_thread_id = GetThreadId((HANDLE)t1);
+   DWORD t2_thread_id = GetThreadId((HANDLE)t2);
+
+   return (t1_thread_id == t2_thread_id) ? EINA_TRUE : EINA_FALSE;
+}
+
+inline Eina_Thread
+_eina_thread_self(void)
+{
+   return (Eina_Thread)GetCurrentThread();
+}
+
+Eina_Bool _eina_thread_name_set(Eina_Thread thread, char *buf)
+{
+   HRESULT res = SetThreadDescription((HANDLE)thread, (PCWSTR)buf);
+   return HRESULT_CODE(res);
+}
+
+Eina_Bool _eina_thread_cancel(Eina_Thread thread)
+{
+   LPDWORD lpExitCode;
+   Eina_Bool success = GetExitCodeThread((HANDLE)thread, lpExitCode);
+   ExitThread(*lpExitCode);
+   return !success;
+}
+void _eina_thread_setcanceltype(int type, int *oldtype)
+{
+   pthread_setcanceltype(EINA_THREAD_CANCEL_DEFERRED, &oldtype);
+}
+int _eina_thread_setcancelstate(int type, int *oldtype)
+{
+   return pthread_setcancelstate(type, &oldtype);
+}
+EAPI Eina_Bool
+_eina_thread_cancellable_set(Eina_Bool cancellable, Eina_Bool *was_cancellable)
+{
+   int state = cancellable ? EINA_THREAD_CANCEL_ENABLE : EINA_THREAD_CANCEL_DISABLE;
+   if (!state)
+   {
+      *was_cancellable = EINA_TRUE;
+      return EINA_FALSE;
+   }
+   else
+   {
+      *was_cancellable = EINA_TRUE;
+      return EINA_TRUE;
+   }
+}
+void _eina_thread_cancel_checkpoint(void)
+{
+   pthread_testcancel();
+}
```

Comment:
Why you have pthread functions at win32 implementation?

Reply:
Just to keep the thread_posix.c default, these functions are not being used and refer to stubs in pthreads.h. are the functions I did not get in win32

Reply:
Maybe is a better idea to include unimplemented.h and mark those as `UNIMPLEMENTED` and add a `#warning` to signalize it